src/units/combine.rs

```diff
@@ -73,10 +91,16 @@ impl Any {
 
                 match res {
                     Ok(update) => {
+                        updates[idx] = Some(update.clone());
                         if Some(idx) == curr_idx {
                             gate.update_data(update.clone()).await;
                         }
-                        updates[idx] = Some(update.clone());
+                        else {
+                            // We currently don’t have an active source but
```

Comment:
I assume thatthe if test can fail for two reasons:
- Some(m) != Some(n)
- Some(m) != None

In the else you don't know which of these was the cause. `curr_idx` could be `Some` (thus an active source) but just a different source than the currently active one, or am I missing something?

Reply:
I think you are right. I added a check to the else branch.